staging/src/k8s.io/apimachinery/pkg/util/strategicpatch/patch_test.go

```diff
@@ -2170,9 +2170,9 @@ mergingList:
 mergingList:
   - name: 2
     other: b
+  - name: 3
   - name: 4
     other: c
-  - name: 3
```

Comment:
/hold

Changes to these test case results represent compatibility breaks we cannot make. Existing API requests generally must produce the same results or we will break existing successful clients

Reply:
I changed this test because of these reasons:
- it seems to describe invalid behavior
- it does not affect content of test data, just changes order of elements after patch

TwoWay patch directive says: add element '3', remove '1', place '2' after '3'.
But in original test we expect new element to be appended to the end of list ignoring order directive. 
```go
			TwoWay: []byte(`
$setElementOrder/mergingList:
  - name: 2
  - name: 3
mergingList:
  - name: 3
  - name: 1
    $patch: delete
`),
			Current: []byte(`
mergingList:
  - name: 1
  - name: 2
    other: b
  - name: 4
    other: c
`),
			Result: []byte(`
mergingList:
  - name: 2
    other: b
  - name: 4
    other: c
  - name: 3
```
Same problem for ThreeWay.

I was able to reproduce original behavior with my new code, but it resulted in breaking all other cases: instead of using order directive it was just appending new elements.
Then i found out why this behavior occurred in this test case: all the code was describing behavior like "keep previous order, then apply order directive for new elements", but side effect on slice`s underlying array was resulting in malformed 'order' slice.

Do we have any common approach for kubernetes to fix such cases? i`m sure its not the first time when we have incorrect behavior described in tests.